sdk/core/Azure.Core/SampleApp/SampleClient.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Azure;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace SampleApp
+{
+    public class SampleClient
+    {
+        private readonly HttpPipeline _pipeline;
+
+        internal static int MyServiceSLAMinutes;
+
+        public SampleClient(Uri endpoint, TokenCredential credential, ClientOptions options)
+        {
+            // Pipeline author can add retry conditions or override the default retry policy
+            HttpPipelineOptions pipelineOptions = new HttpPipelineOptions(options);
+            pipelineOptions.RetryConditions.Add(new GlobalTimeoutRetryCondition(TimeSpan.FromMinutes(MyServiceSLAMinutes)));
+
+            _pipeline = HttpPipelineBuilder.Build(pipelineOptions);
+        }
+
+        public Response GetWithSpecialRetries(RequestContent content, RequestContext context = default)
+        {
+            HttpMessage message = _pipeline.CreateMessage(context);
+
+            // In reality, we would use the protocol method send that takes context, but it's not in Core yet.
+            _pipeline.Send(message, context.CancellationToken);
+            return message.Response;
+        }
+    }
+
+    public class GlobalTimeoutRetryCondition : RetryCondition
+    {
+        private TimeSpan _maxTime;
+
+        public GlobalTimeoutRetryCondition(TimeSpan maxTime)
+        {
+            _maxTime = maxTime;
+        }
+
+        public override bool TryGetShouldRetry(HttpMessage message, out bool shouldRetry)
```

Comment:
The one thing that is missing here from the mutable context approach is the ability for each condition to know what the current state of ShouldRetry is.

Reply:
Yeah, that's not the intention of this pattern.  This pattern basically implements a chain of discrete conditions where any one can decide whether or not to retry, and the first one that does ends the chain.  If different interaction logic between the components being composed is desired, you'd probably want to use a different pattern.  Although, the intention is to make it easy to reason about and debug any issues, so if there's a lot of interaction between logical components, it might be better to go back to the abstract RetryPolicy, and let someone write out the logic in an override of a ShouldRetry() method.

Reply:
Couldn't we pass in the list of conditions and provide a helper method like GetNextShouldRetry?

Reply:
Yeah, I'm open to whatever logic we decide we need for the feature.  We designed the ResponseClassifier chaining pattern to meet the following criteria:
- Each classifier can say whether it does or doesn't "have an opinion" on whether the response is an error (i.e. one that classifies 409 responses doesn't have an opinion on a 404 response) -- that's the "TryGet" semantics
- Having no interactions across classifiers makes it simple to reason about/ no tie-breaking is needed if classifiers disagree
- Precedence is specified via how the chain of classifiers is ordered
- The last item in the chain provides the default classification logic, so you don't end up with no one deciding whether something is an error or not.

In the case you describe above, I'd be curious how you'd break ties if one condition wanted to retry and another didn't, and one didn't want to vote at all if it wasn't relevant to a given response.